lib/sycamore/sycamore/connectors/opensearch/opensearch_reader.py

```diff
@@ -257,15 +257,39 @@ def get_doc_count_for_slice(os_client, slice_query: dict[str, Any]) -> int:
     res = os_client.search(body=slice_query, size=0, track_total_hits=True)
     return res["hits"]["total"]["value"]
 
+def is_compound_query(query: dict[str, Any]) -> bool:
+    """
+    Check if the query is a compound query (i.e., it contains a 'bool' clause).
+    """
+    if "query" not in query:
+        return False
+
+    inner_query = query["query"]
+    if len(inner_query) != 1:
+        return False
+
+    k, _ = next(iter(inner_query.items()))
+    # https://docs.opensearch.org/docs/latest/query-dsl/compound/index/
+    if k in ["bool", "boosting", "constant_score", "dis_max", "function_score", "hybrid"]:
+        return True
+
+    return False
 
 def add_filter_to_query(query: dict[str, Any], filter: dict[str, Any]):
-    actual_query = query["query"]
-    query["query"] = {
-        "bool": {
-            "must": [actual_query],
-            "filter": [{"terms": filter}],
+
+    if is_compound_query(query):
```

Comment:
Per IM discussion, don't do this, instead always wrap -- add a test for the wrapping of a compound query.

https://docs.opensearch.org/docs/latest/query-dsl/compound/index/
says you can have multiple levels "Compound queries serve as wrappers for multiple leaf or compound clauses either to combine their results or to modify their behavior."